controllers/dataplane/openstackdataplanenodeset_controller.go

```diff
@@ -639,9 +646,46 @@ func (r *OpenStackDataPlaneNodeSetReconciler) SetupWithManager(
 			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{})).
 		Watches(&openstackv1.OpenStackVersion{},
 			handler.EnqueueRequestsFromMapFunc(r.genericWatcherFn)).
+		Watches(&machineconfig.MachineConfig{},
+			handler.EnqueueRequestsFromMapFunc(r.machineConfigWatcherFn),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{})).
 		Complete(r)
 }
 
+// machineConfigWatcherFn - watches for changes to the registries MachineConfig resource and queues
+// a reconcile of each NodeSet if the MachineConfig is changed.
+func (r *OpenStackDataPlaneNodeSetReconciler) machineConfigWatcherFn(
+	ctx context.Context, obj client.Object,
+) []reconcile.Request {
+	Log := r.GetLogger(ctx)
+	nodeSets := &dataplanev1.OpenStackDataPlaneNodeSetList{}
+	kind := strings.ToLower(obj.GetObjectKind().GroupVersionKind().Kind)
+	const registryMachineConfigName string = "99-master-generated-registries"
+
+	mcNamespacedName := types.NamespacedName{
+		Name:      registryMachineConfigName,
+		Namespace: "",
+	}
+
+	if err := r.Get(ctx, mcNamespacedName, obj); err != nil {
+		Log.Error(err, fmt.Sprintf("Unable to retrieve MachingConfig %s", registryMachineConfigName))
+		return nil
+	}
+
+	requests := make([]reconcile.Request, 0, len(nodeSets.Items))
```

Comment:
I think we need to query the nodesets first
```suggestion
	listOpts := []client.ListOption{
		client.InNamespace(obj.GetNamespace()),
	}
	if err := r.Client.List(ctx, nodeSets, listOpts...); err != nil {
		Log.Error(err, "Unable to retrieve OpenStackDataPlaneNodeSetList")
		return nil
	}

	requests := make([]reconcile.Request, 0, len(nodeSets.Items))
```